contracts/strategies/aura/base/AuraStrategyUL.sol

```diff
@@ -0,0 +1,613 @@
+  }
+
+  function hodlVault() 
+    public 
+    view 
+    returns (address) 
+  {
+    return getAddress(_HODL_VAULT_SLOT);
+  }
+
+  function depositArbCheck() 
+    public 
+    view 
+    returns(bool) 
+  {
+    return true;
+  }
+
+  function auraPoolId() 
+    public 
+    view 
+    returns (uint256) 
+  {
+    return getUint256(_AURA_POOLID_SLOT);
+  }
+
+  function balancerPoolId() 
+    public 
+    view 
+    returns (bytes32) 
+  {
+    return getBytes32(_BALANCER_POOLID_SLOT);
+  }
+
+  function nTokens() 
+    public 
+    view 
+    returns (uint256) 
+  {
+    return getUint256(_NTOKENS_SLOT);
+  }
+
+  function depositToken() 
+    public 
+    view 
+    returns (address) 
+  {
+    return getAddress(_DEPOSIT_TOKEN_SLOT);
+  }
+
+  function depositReceipt() 
+    public 
+    view 
+    returns (address) 
+  {
+    return getAddress(_DEPOSIT_RECEIPT_SLOT);
+  }
+
+  function depositArrayIndex() 
+    public 
+    view 
+    returns (uint256) 
+  {
+    return getUint256(_DEPOSIT_ARRAY_INDEX_SLOT);
+  }
+
+  function balancerDeposit() 
+    public 
+    view
+    returns (address) 
+  {
+    return getAddress(_BALANCER_DEPOSIT_SLOT);
+  }
+
+  //
+  // Get Strategy Information Functions
+  //
+
+  /** Note that we currently do not have a mechanism here to include the
+   *  amount of reward that is accrued.
+   */
+  function investedUnderlyingBalance() 
+    external 
+    view 
+    returns (uint256) 
+  {
+    return _rewardPoolBalance()
+      .add(IERC20(depositReceipt()).balanceOf(address(this)))
+      .add(IERC20(underlying()).balanceOf(address(this)));
+  }
+
+  //
+  // Strategy Operation Functions - Reinvest
+  //
+
+  /** Get the reward, sell it in exchange for underlying, invest what you got.
+   *  It's not much, but it's honest work.
+   *
+   *  Note that although `onlyNotPausedInvesting` is not added here,
+   *  calling `_investAllUnderlying()` affectively blocks the usage of `doHardWork`
+   *  when the investing is being paused by governance.
+   */
+  function doHardWork() 
+    external 
+    onlyNotPausedInvesting 
+    restricted 
+  {
+    IAuraBaseRewardPool(rewardPool()).getReward();
+    _liquidateReward();
+    _investAllUnderlying();
+  }
+
+  // We assume that all the tradings can be done on Sushiswap
+  function _liquidateReward() 
+    internal 
+  {
+    // Profits can be disabled for possible simplified and rapoolId exit
+    if (!sell()) {
+      emit ProfitsNotCollected(sell(), false);
+      return;
+    }
+
+    for(uint256 i = 0; i < rewardTokens.length; i++){
+      address token = rewardTokens[i];
+      uint256 rewardBalance = IERC20(token).balanceOf(address(this));
+      
+      // if the token is the rewardToken then there won't be a path defined because liquidation is not necessary,
+      // but we still have to make sure that the toHodl part is executed.
+      if (rewardBalance == 0 || 
+          (storedLiquidationDexes[token][rewardToken()].length < 1) && 
```

Comment:
If a certain stored variable, like `rewardToken()` is called more than once within the execution scope, it is more gas efficient to pull it from storage to local memory only once and keep it local for the remainder of the scope. So in this case it would be most efficient to, at the start of this function put `address _rewardToken = rewardToken()` and replace all other cases of `rewardToken()` with `_rewardToken`. Let me know if anything is unclear.